components/layout_2020/flow/inline.rs

```diff
@@ -586,20 +619,19 @@ fn layout_atomic(
     ifc: &mut InlineFormattingContextState,
     atomic: &mut IndependentFormattingContext,
 ) {
-    let style = atomic.style();
+    let style = &atomic.style().clone();
```

Comment:
Could you get away with not cloning the style here? It seems a little expensive.

Reply:
I tried but the borrow checker didn't show mercy :( Will try again.

Reply:
You can avoid the double-borrow here by cloning the position:

```diff
diff --git a/components/layout_2020/flow/inline.rs b/components/layout_2020/flow/inline.rs
index ad1a4f0880..40e562902d 100644
--- a/components/layout_2020/flow/inline.rs
+++ b/components/layout_2020/flow/inline.rs
@@ -619,8 +619,7 @@ fn layout_atomic(
     ifc: &mut InlineFormattingContextState,
     atomic: &mut IndependentFormattingContext,
 ) {
-    let style = &atomic.style().clone();
-    let pbm = style.padding_border_margin(&ifc.containing_block);
+    let pbm = atomic.style().padding_border_margin(&ifc.containing_block);
     let margin = pbm.margin.auto_is(Length::zero);
     let pbm_sums = &(&pbm.padding + &pbm.border) + &margin;
 
@@ -749,8 +748,10 @@ fn layout_atomic(
         block: pbm_sums.block_start,
         inline: ifc.inline_position - ifc.current_nesting_level.inline_start,
     };
-    if style.clone_position().is_relative() {
-        start_corner += &relative_adjustement(&style, ifc.containing_block)
+
+    let position = atomic.style().clone_position();
+    if position.is_relative() {
+        start_corner += &relative_adjustement(atomic.style(), ifc.containing_block)
     }
 
     if let Some(mut child_positioning_context) = child_positioning_context.take() {
```